doc/crypto/api/ops/mac.rst

```diff
@@ -49,6 +51,46 @@ MAC algorithms
 
     The HMAC construction is defined in :RFC-title:`2104`.
 
+    .. _hash-for-hmac:
+
+    .. subsection:: Choice of hash algorithm
+
+        HMAC was designed with hashes that use a Merkle-Damgård construction, for example, MD5, SHA-1, and SHA-2.
+        In particular, the input-block size of the hash must be at least as large as the output size.
+
+        Hash algorithms which have a small input-block size are not directly suitable for use with HMAC.
+        For example, Ascon (`PSA_ALG_ASCON_HASH256`) or Shake-based hashes (`PSA_ALG_SHAKE256_512`).
```

Comment:
[SHAKE](https://en.wikipedia.org/wiki/SHA-3#Instances) does not have a small block size. Quite the opposite, the rate (which is the input-block size for the sponge construction that SHA3 and SHAKE are based on) is larger than for most hashes.

I'm not aware of a technical reason for not using HMAC-SHAKE128 or HMAC-SHAKE256 with the HMAC block size equal to the rate. However, as far as I know, using SHAKE as a hash is something that people only do inside specific algorithms, and I don't know of a good reason not to use one of the actual SHA3 variants if you want HMAC. (Or use KMAC if you want a MAC that's close to SHAKE.)